official/wide_deep/estimator/trainer/model.py

```diff
@@ -0,0 +1,322 @@
+  performed on simpler features. This provides a balance between power and
+  speed that is effective on many structured data problems.
+
+  You can read more about wide and deep models here:
+  https://research.googleblog.com/2016/06/wide-deep-learning-better-together-with.html
+
+  To define model we can use the prebuilt DNNCombinedLinearClassifier class,
+  and need only define the data transformations particular to our dataset, and
+  then
+  assign these (potentially) transformed features to either the DNN, or linear
+  regression portion of the model.
+
+  Args:
+    config: tf.contrib.learn.RunConfig defining the runtime environment for the
+      estimator (including model_dir).
+    embedding_size: int, the number of dimensions used to represent categorical
+      features when providing them as inputs to the DNN.
+    hidden_units: [int], the layer sizes of the DNN (input layer first)
+    learning_rate: float, the learning rate for the optimizer.
+  Returns:
+    A DNNCombinedLinearClassifier
+  """
+  (gender, race, education, marital_status, relationship,
+   workclass, occupation, native_country, age,
+   education_num, capital_gain, capital_loss, hours_per_week) = INPUT_COLUMNS
+  """Build an estimator."""
+
+  # Reused Transformations.
+  # Continuous columns can be converted to categorical via bucketization
+  age_buckets = tf.feature_column.bucketized_column(
+      age, boundaries=[18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  # Wide columns and deep columns.
+  wide_columns = [
+      # Interactions between different categorical features can also
+      # be added as new virtual features.
+      tf.feature_column.crossed_column(
+          ['education', 'occupation'], hash_bucket_size=int(1e4)),
+      tf.feature_column.crossed_column(
+          [age_buckets, race, 'occupation'], hash_bucket_size=int(1e6)),
+      tf.feature_column.crossed_column(
+          ['native_country', 'occupation'], hash_bucket_size=int(1e4)),
+      gender,
+      native_country,
+      education,
+      occupation,
+      workclass,
+      marital_status,
+      relationship,
+      age_buckets,
+  ]
+
+  deep_columns = [
+      # Use indicator columns for low dimensional vocabularies
+      tf.feature_column.indicator_column(workclass),
+      tf.feature_column.indicator_column(education),
+      tf.feature_column.indicator_column(marital_status),
+      tf.feature_column.indicator_column(gender),
+      tf.feature_column.indicator_column(relationship),
+      tf.feature_column.indicator_column(race),
+
+      # Use embedding columns for high dimensional vocabularies
+      tf.feature_column.embedding_column(
+          native_country, dimension=embedding_size),
+      tf.feature_column.embedding_column(occupation, dimension=embedding_size),
+      age,
+      education_num,
+      capital_gain,
+      capital_loss,
+      hours_per_week,
+  ]
+
+  return tf.contrib.learn.DNNLinearCombinedClassifier(
```

Comment:
This has moved to core, so I wonder if we should convert it, or wait for the Experiment equivalent to move.

Reply:
We can't move to core due to the bug b/65479988